src/code/providers/interactive-api-provider.ts

```diff
@@ -148,8 +148,13 @@ class InteractiveApiProvider extends ProviderInterface {
 
   async handleInitialInteractiveState(initInteractiveMessage: IRuntimeInitInteractive) {
     let interactiveState: any
+
+    const initialInteractiveState = initInteractiveMessage.hasLinkedInteractive && initInteractiveMessage.linkedState
+      ? initInteractiveMessage.linkedState
+      : initInteractiveMessage.interactiveState
+
     try {
-      interactiveState = await this.processRawInteractiveState(initInteractiveMessage.interactiveState)
+      interactiveState = await this.processRawInteractiveState(initialInteractiveState)
     }
     catch(e) {
       // on initial interactive state there's not much we can do on error besides ignore it
```

Comment:
I thought about falling back to the interactiveState here if the linkedState has an error but think it would be better to surface that as an error.  I plan on adding that as part of the UI work.